tests/frontend/test_BoardGUI.py

```diff
@@ -1,73 +1,177 @@
-    @patch('pygame.mouse.get_pressed', return_value=(1, 0, 0))  
-    def test_get_cell_clicked(self, mock_get_pressed, mock_get_pos):
-
-        cell = self.board_gui.get_cell_clicked()
-        expected_cell = pg.Vector2(100, 100) // self.board_gui.cell_size
-        self.assertEqual(cell, expected_cell)
-        mock_get_pos.assert_called_once()
-        mock_get_pressed.assert_called_once()
-
-    def test_draw_piece(self):
- 
-        position = (100, 100)
-        self.board_gui.draw_piece('black', position)
-        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.black_piece_image, position)
-
-        self.mock_game.screen.blit.reset_mock()  
-        self.board_gui.draw_piece('white', position)
-        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.white_piece_image, position)
-
-    def test_draw_board(self):
-        self.board_gui.draw_board()
-        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.board_image, (0, 0))
-
-
-if __name__ == '__main__':
-    unittest.main()
+from backend.GameManager import GameManager
+from frontend.BoardGUI import BoardGUI
+from backend.Board import Board
+
+
+@pytest.fixture
+def board_gui_fixture():
+    mock_game = Mock()
+    mock_game.screen = Mock()
+    
+    # Mock Pygame image and transformation methods
+    pg.image.load = Mock()
+    pg.transform.smoothscale = Mock(return_value=Mock())
+    
+    board_size = 8
+    win_size = 640
+    total_pieces = 16
+    
+    board_gui = BoardGUI(mock_game, win_size, board_size, total_pieces)
+    return board_gui, mock_game
+
+
+def test_initialization(board_gui_fixture):
+    board_gui, _ = board_gui_fixture
+    assert board_gui.cell_size == board_gui.win_size // board_gui.board.board_size
+    assert isinstance(board_gui.backend_game, GameManager)
+    assert board_gui.count == 0
+    
+  
+    assert board_gui.game == board_gui_fixture[1]
+
+
+    assert board_gui.board_image is not None
+    assert board_gui.black_piece_image is not None
+    assert board_gui.white_piece_image is not None
+
+
+    assert isinstance(board_gui.board, Board)
+    assert board_gui.board.board_size == 8 # hardecoded above
+
+    assert isinstance(board_gui.backend_game, GameManager)
+    
+    assert board_gui.count == 0
+
+@patch('pygame.image.load')
+@patch('pygame.transform.smoothscale')
+def test_get_scaled_image(mock_smoothscale, mock_load, board_gui_fixture):
+    board_gui, _ = board_gui_fixture
+    mock_image = Mock()
+    mock_load.return_value = mock_image
+    mock_smoothscale.return_value = mock_image
+
+    # Call the method with a mock path and resolution and assert the calls were made correctly
+    path = 'resources/board/board.png'
+    resolution = [100, 100]
+    image = board_gui.get_scaled_image(path, resolution)
+
+    mock_load.assert_called_once_with(path)
+    mock_smoothscale.assert_called_once_with(mock_image.convert_alpha(), resolution)
+    assert image == mock_image
```

Comment:
The true test for get_scaled_image should be that the returned image is the size of the resolution provided to the function.

We would also want a test for a negative path. If image.get_size() is not equal to resoultion then scaled_image.get_size() doesnt equal image.get_size()

Also Im not sure if you need to mock out the image and function here. Since we have the actual image path we can use and we can use the pygame.transform.smoothscale I think we can just use the actual artifacts to run the test without any mocks.